pyls/server.py

```diff
@@ -60,32 +74,59 @@ def _content_length(self, line):
             except ValueError:
                 raise ValueError("Invalid Content-Length header: {}".format(value))
 
-    def _read_message(self):
+    def readline(self):
         line = self.rfile.readline()
 
+        if isinstance(line, bytes):
+            line = line.decode('utf-8')
+
+        return line
+
+    def _read_message(self):
+        line = self.readline()
+
         if not line:
             raise EOFError()
 
         content_length = self._content_length(line)
 
         # Blindly consume all header lines
         while line and line.strip():
-            line = self.rfile.readline()
+            cl = self._content_length(line)
+            if cl:
+                content_length = cl
+            line = self.readline()
 
         if not line:
             raise EOFError()
 
         # Grab the body
-        return self.rfile.read(content_length)
+        res = self.rfile.read(content_length)
+
+        if isinstance(res, bytes):
+            res = res.decode('utf-8')
+
+        return res
 
     def _write_message(self, msg):
         body = json.dumps(msg, separators=(",", ":"))
         content_length = len(body)
+
+        if isinstance(self.wfile, socket_type):
+            header = "HTTP/1.1 200 OK\r\n"
```

Comment:
This isn't HTTP, not sure we should be adding this header just so a requests-based client works